<launch>
	<!-- LAUNCH INTERFACE -->
	<!-- specify whether the arduino board is connected or not -->
	<arg name="isBoardConnected" default="true"/>
	<arg name="hand_name" default="soft_hand"/>
	<arg name="port" default="/dev/ttyACM0"/>
	<arg name="baud" default="115200"/>

	<!-- LAUNCH IMPLEMENTATION -->
	<!-- remember to set the parameters properly -->

	<param name="hand_name" value="$(arg hand_name)"/>

	<node name="flexiforce_publisher" pkg="flexiforce_glove" type="flexiforce_joint_state_publisher" respawn="false" output="screen"/> 

	<!-- launch the arduino USB connection if connected -->
	<group if="$(arg isBoardConnected)">
		<rosparam command="load" file="$(find flexiforce_glove)/config/flexiforce_setup.yaml" />
		<node name="arduino_node" pkg="rosserial_python" type="serial_node.py">
			<param name="~port" value="$(arg port)" />
	        <param name="~baud" value="$(arg baud)" />
		</node>
	</group>

</launch>